ts/src/header-validator/validate-json.ts

```diff
@@ -891,6 +913,34 @@ function channelCapacity(ctx: SourceContext, s: Source): void {
   }
 }
 
+function validateAttributionScopeFields(ctx: SourceContext, s: Source): boolean {
+  if (s.attributionScopeLimit === null) {
+    if (s.attributionScopes !== null || s.maxEventStates !== null) {
+    	ctx.error(
+      	  'attribution scope limit should be set if attribution scopes or max event states are set')
+      return false
+    }
+    return true
+  }
+  if (s.attributionScopeLimit < 1) {
```

Comment:
`uint32` doesn't check if it's > 0, right?

Reply:
Ah. I'd suggest not using `uint32` directly with that field then, and instead adding a helper function that also does the range check, so that the `> 0` condition is checked in the proper location and results in an error with the proper path if it isn't met.

Reply:
Good point. Done.